packages/types/index.d.ts

```diff
@@ -0,0 +1,57 @@
+// index.d.ts
+
+export type Input =
+  | {
+      prompt: string;
+    }
+  | {
+      messages: ChatMessage[];
+    };
+
+export type Output =
+  | {
+      text: string;
+    }
+  | {
+      message: ChatMessage;
+    };
+
+type LLM =
+  | "openai/gpt3.5"
+  | "together/gpt-neoxt-20B"
+  | "cohere/xlarge"
+  | "local";
+
+export interface CompletionOptions {
+  onStreamResult?: (result: Output | null, error: string | null) => unknown;
```

Comment:
there are lots of comments on these options in the `public-interface.ts` file - wondering if we can preserve those so VSCode shows them when you're coding?